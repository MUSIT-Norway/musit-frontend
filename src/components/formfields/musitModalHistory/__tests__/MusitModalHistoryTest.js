/**
 * Created by steinaol on 24.10.16.
 */

import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';
import React from 'react'
import MusitModalHistory from '../index';

describe('MusitModalHistory', () => {
    const testData =
        [
            {
                doneBy: 'Kjell Kjellstad',
                doneDate: '23.12.2001',
                name: 'Arne And',
                from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
                to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
            },
            {
                doneBy: 'Bjørn Bjørnsen',
                doneDate: '21.12.2005',
                name: 'Arne And',
                from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            },
            {
                doneBy: 'Jan Pettersen',
                doneDate: '12.01.2015',
                name: 'Arne And',
                from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            },
            {
                doneBy: 'Kjell Kjellstad',
                doneDate: '23.12.2001',
                name: 'Arne And',
                from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
                to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
            },
            {
                doneBy: 'Bjørn Bjørnsen',
                doneDate: '21.12.2005',
                name: 'Arne And',
                from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            },
            { doneBy: 'Jan Pettersen',
                doneDate: '12.01.2015',
                name: 'Arne And',
                from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            },
            { doneBy: 'Bjørn Bjørnsen',
                doneDate: '21.12.2005',
                name: 'Arne And',
                from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            },
            { doneBy: 'Kjell Kjellstad',
                doneDate: '23.12.2001',
                name: 'Arne And',
                from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
                to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
            },
            { doneBy: 'Bjørn Bjørnsen',
                doneDate: '21.12.2005',
                name: 'Arne And',
                from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
                to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
            }
    ]

    const object={
        id: 1,
        museumNo: 'TRH/V/233',
        subNo: '0',
        term: 'Carex saxatilis'
    }
    it('should display object 1', () => {
        const wrapper = shallow(
            <MusitModalHistory
                moves={testData}
                show
                headerText="Flyttehistorikk"
                onClose={() => true}
                onHide={() => true}
                object={object}
                translate={() => true }
            />
        )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})
