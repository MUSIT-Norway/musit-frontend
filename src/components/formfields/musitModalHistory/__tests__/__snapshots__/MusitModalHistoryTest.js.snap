exports[`MusitModalHistory should display object 1 1`] = `
<div>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-sm"
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    bsSize="large"
    dialogClassName="my-modal"
    dialogComponentClass={[Function ModalDialog]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "modals": Array []
      }
    }
    onHide={[Function onHide]}
    renderBackdrop={[Function renderBackdrop]}
    show={true}>
    <ModalHeader
      aria-label="Close"
      bsClass="modal-header"
      closeButton={true}
      style={
        Object {
          "border": "none"
        }
      }>
      <ModalTitle
        bsClass="modal-title"
        id="title"
        style={
          Object {
            "textAlign": "center"
          }
        }>
        Flyttehistorikk TRH/V/233 - 0 - Carex saxatilis
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      style={
        Object {
          "height": 300,
          "overflow": "auto"
        }
      }>
      <ModalMoveHistoryGrid
        tableData={
          Array [
            Object {
              "doneBy": "Kjell Kjellstad",
              "doneDate": "23.12.2001",
              "from": Object {
                "path": "NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8"
              }
            },
            Object {
              "doneBy": "Bjørn Bjørnsen",
              "doneDate": "21.12.2005",
              "from": Object {
                "path": "NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            },
            Object {
              "doneBy": "Jan Pettersen",
              "doneDate": "12.01.2015",
              "from": Object {
                "path": "KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            },
            Object {
              "doneBy": "Kjell Kjellstad",
              "doneDate": "23.12.2001",
              "from": Object {
                "path": "NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8"
              }
            },
            Object {
              "doneBy": "Bjørn Bjørnsen",
              "doneDate": "21.12.2005",
              "from": Object {
                "path": "NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            },
            Object {
              "doneBy": "Jan Pettersen",
              "doneDate": "12.01.2015",
              "from": Object {
                "path": "KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            },
            Object {
              "doneBy": "Bjørn Bjørnsen",
              "doneDate": "21.12.2005",
              "from": Object {
                "path": "NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            },
            Object {
              "doneBy": "Kjell Kjellstad",
              "doneDate": "23.12.2001",
              "from": Object {
                "path": "NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8"
              }
            },
            Object {
              "doneBy": "Bjørn Bjørnsen",
              "doneDate": "21.12.2005",
              "from": Object {
                "path": "NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10"
              },
              "name": "Arne And",
              "to": Object {
                "path": "NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3"
              }
            }
          ]
        }
        translate={[Function translate]} />
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      style={
        Object {
          "textAlign": "center"
        }
      }>
      <br />
      <Row
        bsClass="row"
        componentClass="div"
        style={
          Object {
            "textAlign": "center"
          }
        }>
        <Col
          bsClass="col"
          componentClass="div"
          sm={4}
          smOffset={4}
          xs={4}>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[Function onClose]}>
            Lukk
          </Button>
        </Col>
      </Row>
    </ModalFooter>
  </Modal>
</div>
`;
